### ISSUE TITLE
Null pointer when clicking on the background of the feature diagram

### ISSUE THREAD

**Issue by tthuem** [FeatureIDE/FeatureIDE#703]
#### Prerequisitives
* **Eclipse Version:**  4.7.2
* **FeatureIDE-Version:**  release branch
* **Operating system:** Win10

### Issue description
Clicking in the feature diagram on the white background gives an exception in the error log. However, I did not face any further problems besides entry in the error log. Does not pop-up to the user.

```
eclipse.buildId=4.7.2.M20171130-0510
java.version=1.8.0_161
java.vendor=Oracle Corporation
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=de_DE
Framework arguments:  -product org.eclipse.platform.ide
Command-line arguments:  -product org.eclipse.platform.ide -data C:\Users\tthuem\Tools\workspace-4.7/../workspace-4.7-runtime -dev file:C:/Users/tthuem/Tools/workspace-4.7/.metadata/.plugins/org.eclipse.pde.core/FeatureIDE/dev.properties -os win32 -ws win32 -arch x86_64 -consoleLog

org.eclipse.ui
Error
Thu Feb 08 16:26:12 CET 2018
Unhandled event loop exception

java.lang.NullPointerException
	at de.ovgu.featureide.fm.ui.editors.elements.GraphicalFeature.toString(GraphicalFeature.java:169)
	at java.lang.String.valueOf(Unknown Source)
	at java.lang.StringBuilder.append(Unknown Source)
	at java.util.AbstractMap.toString(Unknown Source)
	at de.ovgu.featureide.fm.ui.editors.elements.GraphicalFeatureModel.toString(GraphicalFeatureModel.java:229)
	at de.ovgu.featureide.fm.ui.editors.featuremodel.actions.colors.SetFeatureColorAction.isSelectionValid(SetFeatureColorAction.java:146)
	at de.ovgu.featureide.fm.ui.editors.featuremodel.actions.colors.SetFeatureColorAction$1.selectionChanged(SetFeatureColorAction.java:82)
	at org.eclipse.gef.ui.parts.AbstractEditPartViewer.fireSelectionChanged(AbstractEditPartViewer.java:247)
	at org.eclipse.gef.ui.parts.AbstractEditPartViewer$1.run(AbstractEditPartViewer.java:131)
	at org.eclipse.gef.SelectionManager.fireSelectionChanged(SelectionManager.java:156)
	at org.eclipse.gef.SelectionManager.setSelection(SelectionManager.java:314)
	at org.eclipse.gef.ui.parts.AbstractEditPartViewer.setSelection(AbstractEditPartViewer.java:751)
	at org.eclipse.gef.tools.MarqueeSelectionTool.performMarqueeSelect(MarqueeSelectionTool.java:744)
	at org.eclipse.gef.tools.MarqueeSelectionTool.handleButtonUp(MarqueeSelectionTool.java:487)
	at org.eclipse.gef.tools.AbstractTool.mouseUp(AbstractTool.java:1200)
	at org.eclipse.gef.tools.SelectionTool.mouseUp(SelectionTool.java:574)
	at org.eclipse.gef.EditDomain.mouseUp(EditDomain.java:301)
	at org.eclipse.gef.ui.parts.DomainEventDispatcher.dispatchMouseReleased(DomainEventDispatcher.java:380)
	at org.eclipse.draw2d.LightweightSystem$EventHandler.mouseUp(LightweightSystem.java:548)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:221)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4428)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1079)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4238)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3817)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1150)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1039)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:680)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:594)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:151)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:653)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:590)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1499)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1472)
```

Was only able to reproduce it with the following model in a feature modeling project:

```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
	<featureModel>
		<properties/>
		<struct>
			<alt mandatory="true" name="VW">
				<feature name="Up"/>
				<feature name="Polo"/>
				<alt name="Golf">
					<and name="Trendline">
						<alt mandatory="true" name="Motor">
							<alt name="Benzin">
								<feature name="1.0 TSI 5-Gang"/>
								<feature name="1.0 TSI 6-Gang"/>
								<feature name="1.0 TSI 7-Gang"/>
							</alt>
							<alt name="Diesel">
								<feature name="1.6 TDI 5-Gang"/>
								<feature name="1.6 TDI 7-Gang"/>
							</alt>
						</alt>
						<feature mandatory="true" name="Außenlackierung ..."/>
						<feature mandatory="true" name="Felgen ..."/>
						<feature mandatory="true" name="Interieur ..."/>
						<and mandatory="true" name="Ausstattung">
							<feature name="Business-Paket">
								<description>
									Mittelarmlehne vorn mit Ablagebox und 2 Ausströmern hinten
									    Scheibenwaschdüsen vorn automatisch beheizt
									    Scheinwerfer-Reinigungsanlage
									    Vordersitze beheizbar
									    Warnleuchte für Waschwasserstand
									
									braucht
									Navigationsfunktion "Discover Media" (für "Composition Media")
									oder
									Radio "Composition Media"
								</description>
							</feature>
							<feature name="4 Türen"/>
							<feature name="Airbags">
								<description>
									braucht
									Proaktives Insassenschutzsystem
									oder
									Proaktives Insassenschutzsystem in Verbindung mit "Front Assist"
								</description>
							</feature>
							<feature name="Anhängevorrichtung anklappbar"/>
							<feature name="Anschlussgarantie ..."/>
							<and name="Car-Net">
								<feature name="Guide &amp; Inform Laufzeit 1 Jahr">
									<description>
										braucht
										Navigationsfunktion "Discover Media" (für "Composition Media")
									</description>
								</feature>
								<feature name="Guide &amp; Inform und Security &amp; Service Plus Laufzeit 1 Jahr inkl. Security &amp; Service Basic Laufzeit 10 Jahre">
									<description>
										braucht
										Navigationsfunktion "Discover Media" (für "Composition Media")
									</description>
								</feature>
								<feature name="Security &amp; Service Plus Laufzeit 1 Jahr inkl. Security &amp; Service Basic Laufzeit 10 Jahre">
									<description>
										braucht
										Notruf-Service, Laufzeit 10 Jahre; keine Registrierung notwendig, Dienst ist bei Auslieferung aktiviert
									</description>
								</feature>
							</and>
						</and>
					</and>
					<feature name="Comfortline"/>
					<feature name="Highline"/>
					<feature name="GTE"/>
					<feature name="GTD"/>
					<feature name="GTI"/>
					<feature name="GTI Performance"/>
					<feature name="R"/>
					<feature name="e-Golf"/>
					<feature name="TGI Trendline"/>
					<feature name="TGI Comfortline"/>
					<feature name="Join"/>
				</alt>
				<feature name="T-Roc"/>
				<feature name="Golf Sportsvan"/>
				<feature name="Golf Variant"/>
				<feature name="Beetle Cabriolet"/>
				<feature name="Touran"/>
				<feature name="Tiguan"/>
				<feature name="Tiguan Allspace"/>
				<feature name="Passat"/>
				<feature name="Passat Variant"/>
				<feature name="Arteon"/>
				<feature name="Sharan"/>
				<feature name="Touareg"/>
				<feature name="Nutzfahrzeuge"/>
			</alt>
		</struct>
		<constraints/>
		<calculations Auto="true" Constraints="true" Features="true" Redundant="true" Tautology="true"/>
		<comments/>
		<featureOrder userDefined="false"/>
	</featureModel>
```

**Comment by tthuem** [comment]
Who can have a look at this?

**Comment by Subaro** [comment]
fixed with 62258596f7c63307949a80bb926efbd6314e330d and the other one above

**Comment by tthuem** [comment]
The problem does not occur anymore. Thanks!
